darwin/future/meta/objects/stage.py

```diff
@@ -75,6 +83,30 @@ def move_attached_files_to_stage(self, new_stage_id: UUID) -> Stage:
             self.meta_params["dataset_id"],
         )
         ids = [x.id for x in self.item_ids.collect_all()]
+
+        def all_items_complete(item_ids: list[UUID]) -> bool:
+            for item_id in item_ids:
+                if get_item(self.client, slug, item_id).processing_status != "complete":
+                    return False
+            return True
+
+        if wait:
+            max_attempts = 5
+            wait_time = 0.5  # seconds
```

Comment:
Magic numbers might be a good idea to abstract into optional function parameters with defaults.

Reply:
I agree - adding these as default arguments in the top level(s) of this function calls in a new commit.